o Don't deny access to everyone if the calling application didn't set a TTY;
  we wouldn't do it for unprivileged users, regardless, so don't lock them out.

--- Linux-PAM-0.77/modules/pam_securetty/pam_securetty.c	2002-07-13 01:10:54.000000000 -0400
+++ Linux-PAM-0.77/modules/pam_securetty/pam_securetty.c	2003-07-01 19:49:01.000000000 -0400
@@ -97,6 +97,15 @@
 	return (retval == PAM_CONV_AGAIN ? PAM_INCOMPLETE:PAM_SERVICE_ERR);
     }
 
+    user_pwd = getpwnam(username);
+    if (user_pwd == NULL) {
+	return PAM_IGNORE;
+    } else if (user_pwd->pw_uid != 0) { /* If the user is not root,
+					   securetty's does not apply
+					   to them */
+	return PAM_SUCCESS;
+    }
+
     retval = pam_get_item(pamh, PAM_TTY, (const void **)&uttyname);
     if (retval != PAM_SUCCESS || uttyname == NULL) {
         if (ctrl & PAM_DEBUG_ARG) {
@@ -110,15 +119,6 @@
 	uttyname += sizeof(TTY_PREFIX)-1;
     }
 
-    user_pwd = getpwnam(username);
-    if (user_pwd == NULL) {
-	return PAM_IGNORE;
-    } else if (user_pwd->pw_uid != 0) { /* If the user is not root,
-					   securetty's does not apply
-					   to them */
-	return PAM_SUCCESS;
-    }
-
     if (stat(SECURETTY_FILE, &ttyfileinfo)) {
 	_pam_log(LOG_NOTICE, "Couldn't open " SECURETTY_FILE);
 	return PAM_SUCCESS; /* for compatibility with old securetty handling,
